test/extensions/filters/listener/original_dst/original_dst_integration_test.cc

```diff
@@ -0,0 +1,134 @@
+    RELEASE_ASSERT(bootstrap.mutable_static_resources()->clusters_size() == 1, "");
+    auto& cluster = *bootstrap.mutable_static_resources()->mutable_clusters(0);
+    cluster.set_type(envoy::config::cluster::v3::Cluster::ORIGINAL_DST);
+    cluster.set_lb_policy(envoy::config::cluster::v3::Cluster::CLUSTER_PROVIDED);
+    cluster.mutable_original_dst_lb_config()->mutable_upstream_port_override()->set_value(port);
+    cluster.clear_load_assignment();
+
+    auto* listener = bootstrap.mutable_static_resources()->mutable_listeners(0);
+    listener->mutable_address()->mutable_socket_address()->set_address("0.0.0.0");
+
+    auto* listener_filter = listener->add_listener_filters();
+    listener_filter->set_name("envoy.filters.listener.original_dst");
+    envoy::extensions::filters::listener::original_dst::v3::OriginalDst original_dst;
+    listener_filter->mutable_typed_config()->PackFrom(original_dst);
+  };
+}
+
+INSTANTIATE_TEST_SUITE_P(Protocols, OriginalDstIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams(
+                             {Http::CodecType::HTTP1}, {Http::CodecType::HTTP1})),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+TEST_P(OriginalDstIntegrationTest, OriginalDstHttpManyConnections) {
+  // Only do this for IPv4 as we can have many 127.0.0.x addresses listening on 0.0.0.0
+  if (version_ != Network::Address::IpVersion::v4) {
+    return;
+  }
+  auto address = Network::Test::getAnyAddress(Network::Address::IpVersion::v4);
+  Network::DownstreamTransportSocketFactoryPtr factory =
+      upstream_tls_ ? createUpstreamTlsContext(upstreamConfig())
+                    : Network::Test::createRawBufferDownstreamSocketFactory();
+  fake_upstreams_.emplace_back(std::make_unique<FakeUpstream>(std::move(factory), address,
+                                                              configWithType(upstreamProtocol())));
+
+  config_helper_.addConfigModifier(
+      setOriginalDstCluster(fake_upstreams_[0]->localAddress()->ip()->port()));
+  initialize();
+
+  const int32_t kMaxConnections = 100;
+  for (int i = 0; i < kMaxConnections; ++i) {
+    Network::ClientConnectionPtr connection(dispatcher_->createClientConnection(
+        Network::Utility::resolveUrl(fmt::format("tcp://127.0.0.{}:{}", i + 1, lookupPort("http"))),
+        Network::Address::InstanceConstSharedPtr(), Network::Test::createRawBufferSocket(), nullptr,
+        nullptr));
+
+    connection->enableHalfClose(enableHalfClose());
+
+    codec_client_ = makeHttpConnection(std::move(connection));
+    auto response =
+        sendRequestAndWaitForResponse(default_request_headers_, 0, default_response_headers_, 0);
+    ASSERT_TRUE(fake_upstream_connection_->close());
+    fake_upstream_connection_.reset();
+    codec_client_->close();
+  }
+}
+
+class OriginalDstTcpProxyIntegrationTest
+    : public testing::TestWithParam<Network::Address::IpVersion>,
```

Comment:
There is check in the test, it will return directly when the test isn't ipv4, then why we still add param for the test?